### PR TITLE
Add badge to topology labels

### PR DESCRIPTION
**Pull request by sahil143**
ODC Story: https://jira.coreos.com/browse/ODC-1192
This Pr adds the Badge to topology labels.

image

UX:
image

### CODE REVIEW

**Comment by christianvogt** [comment]
@sahil143  Why are you not using `ResourceIcon` from `public/components/utils/resource-icon.tsx` and embedding it as a foreign object? You should also be calculating the amount of width it consumes so that the padding is equivalent on both sides. As seen in your two screenshots the padding is inconsistent.

**Comment by christianvogt** [comment]
These look too different:
image

`ResourceIcon` has `7px` padding and `20px` border radius.

File: frontend/packages/dev-console/src/components/topology/shapes/Badge.scss [hunk 1]
```diff
@@ -0,0 +1,11 @@
+.odc-badge-icon {
+  fill: var(--pf-global--info-color--200);
+  text {
+    fill: #fff !important;
+  }
+}
+
+.odc-badge-deployment,
+.odc-badge-deploymentconfig {
+  fill: var(--pf-global--info-color--200);
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/topology/shapes/Badge.scss]: argh more duplication.

ResourceIcon makes use of this:
```
.co-m-resource-daemonset,
.co-m-resource-deployment,
.co-m-resource-deploymentconfig,
.co-m-resource-job,
.co-m-resource-petset,
.co-m-resource-replicaset,
.co-m-resource-replicationcontroller {
  background-color: $color-pod-overlord;
}
```

File: frontend/packages/dev-console/src/components/svg/SvgBoxedText.tsx [hunk 2]
```diff
@@ -55,24 +59,36 @@ export default class SvgBoxedText extends React.PureComponent<SvgBoxedTextProps,
...
+            x={x - paddingX - bb.width / 2 - badgeSpace / 2}
+            width={bb.width + paddingX * 2 + badgeSpace}
             y={y - paddingY - bb.height / 2}
             height={bb.height + paddingY * 2}
             rx={cornerRadius}
             ry={cornerRadius}
           />
         )}
-        <text {...other} ref={this.textRef} x={x} y={y} textAnchor="middle" dy="0.3em">
+        {bb && kind && <Badge ref={this.badgeRef} x={x - bb.width / 2 - 2} y={y} kind={kind} />}
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/svg/SvgBoxedText.tsx]: magic number `- 2` ?

File: frontend/packages/dev-console/src/components/topology/shapes/Badge.scss [hunk 3]
```diff
@@ -0,0 +1,11 @@
+.odc-badge-icon {
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/topology/shapes/Badge.scss]: Convention would be:
```suggestion
.odc-badge {
```

File: frontend/packages/dev-console/src/components/topology/shapes/Badge.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+export function getKindStringAndAbbrivation(kind: string) {
+  const kindObj = modelFor(kind);
+  const kindStr = get(kindObj, 'kind', kind);
+  const KindAbbr = (kindObj && kindObj.abbr) || kindToAbbr(kindStr);
+  return { kindStr, KindAbbr };
+}
+
+export const Badge: React.FC<BadgeProps> = (
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/topology/shapes/Badge.tsx]: `Badge` is very generic. Maybe also call it `ResourceIcon` ?

File: frontend/packages/dev-console/src/components/topology/shapes/Badge.scss (hunk 1 above)
Comment (christianvogt) [reply frontend/packages/dev-console/src/components/topology/shapes/Badge.scss]: We also support daemonsets and therefore need it's color set.

File: frontend/packages/dev-console/src/components/svg/SvgBoxedText.tsx (hunk 2 above)
Comment (sahil143) [reply frontend/packages/dev-console/src/components/svg/SvgBoxedText.tsx]: 2 is the space b/w label and badge.

**Comment by sahil143** [comment]
>These look too different:
image
ResourceIcon has 7px padding and 20px border radius

Currently, the size of the background rectangle is being calculated based on the text width. Should we make rect a fixed width?

**Comment by sahil143** [comment]
@christianvogt I have made the requested changes. also changed the padding and border-radius.
Screenshot from 2019-07-18 09-49-09

Also updated the screenshots in the description.

**Comment by christianvogt** [comment]
@sahil143 create a PR against your PR to try and tinker with this a bit. See https://github.com/sahil143/console/pull/1

**Pull request merged**